kernel32.cpp

```diff
@@ -437,6 +437,30 @@ namespace kernel32 {
 		return r;
 	}
 
+	int WIN_FUNC SetEndOfFile(void *hFile) {
+		DEBUG_LOG("SetEndOfFile\n");
+		FILE *fp = files::fpFromHandle(hFile);
+		return ftruncate(fileno(fp), ftell(fp)) == 0;
```

Comment:
fflush(fp) before to make sure there's nothing cached?

Reply:
Is it necessary to call flush first? I thought it would happen automatically since the file API is supposed to be synchronous.

Reply:
I believe so. https://stackoverflow.com/questions/2806104/does-fwrite-buffer-the-output

Reply:
Well, that's for fwrite, not ftruncate, but I'll do the flush just to be sure.

Reply:
The point is that fwrite may buffer, while ftruncate operates on file descriptors which is the layer below file handles, so it won't flush those buffers before running. `ftell` may, but I wouldn't rely on that.